### DISCUSSION TITLE
`sndfile` library not found when running actions on `ubuntu-latest`

### DISCUSSION THREAD

**Discussion by ye-yu** [community/community#25532] [Actions]
<p>I perform python unit tests in my GitHub Actions. This is my first action, and the operation stopped on the first run because <code>sndfile</code> library is not found in the system. This is the full log trace:</p>
<pre><code>Run flake8 aaapi tests
running test
running egg_info
creating aaapi.egg-info
writing aaapi.egg-info/PKG-INFO
writing dependency_links to aaapi.egg-info/dependency_links.txt
writing top-level names to aaapi.egg-info/top_level.txt
writing manifest file 'aaapi.egg-info/SOURCES.txt'
reading manifest file 'aaapi.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching ' __pycache__' found under directory '*'
warning: no previously-included files matching '*.py[co]' found under directory '*'
warning: no files found matching '*.jpg' under directory 'docs'
warning: no files found matching '*.png' under directory 'docs'
warning: no files found matching '*.gif' under directory 'docs'
writing manifest file 'aaapi.egg-info/SOURCES.txt'
running build_ext
error: sndfile library not found
</code></pre>
<p>This is likely due to the packages that I used in my python code, and I found out that the system need the package <code>libsndfile1-dev</code> installed in the ubuntu system. I assume that I cannot do this because I probably do not have the root access to the container of the action. How do I resolve this? Thank you.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>GitHub-hosted runners has administrative privileges.</p>
<p>The Linux and macOS virtual machines both run using passwordless sudo. When you need to execute commands or install tools that require more privileges than the current user, you can use sudo without needing to provide a password.</p>
<p>You can refer to <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/virtual-environments-for-github-hosted-runners#administrative-privileges-of-github-hosted-runners" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/virtual-environments-for-github-hosted-runners#administrative-privileges-of-github-hosted-runners</a></p>
<p>&gt;&gt;  <em>I assume that I cannot do this because I probably do not have the root access to the container of the action.</em></p>
<p>Wht’s the meaning of <em>the container of the action?</em> Could you please show your workflow yml content? </p>
<p>I tested in my side, I can use command <strong>sudo apt-get install libsndfile1-dev</strong> to install libsndfile1-dev to the hosted ubuntu runner machine .</p>
<p>name: install sndfile</p>
<p>on: push</p>
<p>jobs:</p>
<p>  job1:</p>
<p>    runs-on: ubuntu-latest</p>
<p>    steps:</p>
<p>     - run: |</p>
<p>        sudo apt-get install libsndfile1-dev</p>
